### PR TITLE
RASS-586 500 error when trying to submit offence without overall sentence lengh

### PR DESCRIPTION
**Pull request by pavilionsahota**

### CODE REVIEW

File: server/views/pages/offence/check-offence-answers.njk [hunk 1]
```diff
@@ -75,7 +75,7 @@
                 convictionDate: offence.sentence.convictionDate | formatDate,
                 terrorRelated: offence.terrorRelated,
                 isSentenced: offence.sentence,
-                custodialSentenceLength: offence.sentence.periodLengths | findPeriodLengthByType("SENTENCE_LENGTH"),
+                custodialSentenceLength: offence.sentence.periodLengths | findPeriodLengthByType("SENTENCE_LENGTH") if courtAppearance.hasOverallSentenceLength === 'true',
```
Comment (carloveo-moj) [inline server/views/pages/offence/check-offence-answers.njk:78]: this still needs to be shown regardless of an overall sentence length doesn't it?

Comment (pavilionsahota) [reply server/views/pages/offence/check-offence-answers.njk:78]: It was causing an error trying to render this template. Should there be a default value like '_Not entered_' or something?

Comment (carloveo-moj) [reply server/views/pages/offence/check-offence-answers.njk:78]: it shouldn't show if there isn't a value for SENTENCE_LENGTH, what is the error? I thought the if statement https://github.com/ministryofjustice/hmpps-remand-and-sentencing/blob/main/server/utils/nunjucksSetup.ts#L110 managed a null period lengths?

**Comment by ldlharper** [comment]
I think this can be fixed by changing CalculateReleaseDatesService.compareOverallSentenceLength

There is a check to see if the overall sentence length is set:
```
    if (sentences.length && appearance.hasOverallSentenceLength) {
```

I didn't realise it was a string, not a boolean. It needs to be:

```
    if (sentences.length && appearance.hasOverallSentenceLength === 'true') {
```

File: server/views/pages/offence/check-offence-answers.njk (hunk 1 above)
Comment (pavilionsahota) [reply server/views/pages/offence/check-offence-answers.njk]: You're right, the fix was the `=== 'true'` I added to Line 24. Not this one. I've removed this and have pushed up

**Review by carloveo-moj — APPROVED** [review]
LGTM

**Pull request merged**